department-of-veterans-affairs/caseflow
# adding logic to bring in issue-modification request and updating the …

Resolves [APPEALS-45332](https://jira.devops.va.gov/browse/APPEALS-45332)

# Description
Please explain the changes you made here.
- Added logic to bring request issue modification in the csv, 
- event type filter and pending status filter was added. 
- spec was written to test the feature. 
- 

## Acceptance Criteria
- Generate Task Report should be working.
- Generate Task Report when Report Type -> event/Action is selected then it should return all the event with pending status / request issue modification and approval if they exist in the system 
- should be able to filter newly added event type and status. 

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
2. Login as an admin -> VHAAdmin or VHAADMIN2 
3. Go to -> /decision_reviews/vha/report 
4. Select Event/Action in the report type. 
5. Click on the Generate Task report -> it should generate a CSV file. 
6. Verify if there are any pending status row in the CSV file. 
7. if not present ->  go through an intake progress as a none-admin, make a request addition/ modification request- save. 
8. as a none admin - Edit/ cancel or withdraw different request modification you made. 
9. Login as a admin, approve the request modification from pending tab. 
10. Go back to /decision_reviews/vha/report and generate report.
11. You should see an event for all of these action you took as a non-admin and admin. 



# Frontend
## No frontend facing changes are present. 

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
